### PR TITLE
Support terminal

### PR DESCRIPTION
**Pull request by oeway**
This is the terminal backend for https://github.com/oeway/ImJoy/pull/146

### CODE REVIEW

File: imjoy/imjoyPluginEngine.py [hunk 1]
```diff
@@ -731,6 +737,102 @@ def connect(sid, environ):
...
+            cmd = " ".join(shlex.quote(c) for c in cmd)
+            print("child pid is", child_pid)
+            print(
+                f"starting background task with command `{cmd}` to continously read "
+                "and forward pty output to client"
+            )
+            logger.debug("xterm task started", terminal_session)
+            # os.write(terminal_session["fd"], "\r".encode())
+            asyncio.ensure_future(
+                read_and_forward_terminal_output(), loop=asyncio.get_event_loop()
```
Comment (MartinHjelmare) [inline imjoy/imjoyPluginEngine.py]: When is the task cancelled?

Comment (oeway) [reply imjoy/imjoyPluginEngine.py]: It will start forever. Will fix when we support multiple terminal sessions.

Comment (MartinHjelmare) [reply imjoy/imjoyPluginEngine.py]: We don't need to clean up tasks when the plugin exits?

Comment (oeway) [reply imjoy/imjoyPluginEngine.py]: This is not a plugin, it's an engine service, which shares the life cycle with the engine itself.

Comment (oeway) [reply imjoy/imjoyPluginEngine.py]: I think we will have trouble when the terminal process accidentally closed. But let's leave it for now. We will handle it when we start to work on the multi-terminal session support.

**Pull request merged**